packages/sit-onyx/src/components/TestInput/TestInput.vue

```diff
@@ -31,15 +66,48 @@ const handleChange = (event: Event) => {
   const target = event.target as HTMLInputElement;
   emit("change", target.value);
 };
+
+onMounted(() => {
+  /* element will always be truthy onMounted */
+  const element = coreElement.value;
+  if (element) {
+    watch(
+      value,
+      () => {
+        /* we need to destructure so that the emitter below fires properly. */
+        validityState.value = { ...element.validity };
+      },
+      { immediate: true },
+    );
+    watch(
+      validityState,
+      () => {
+        if (element.validity) {
+          emit("validityChange", element.validity);
+        }
+      },
+      { deep: true },
+    );
+  }
+});
```

Comment:
```suggestion
watch([value, coreElement], () => {
  // update validity state when value changes
  if (!coreElement.value) return;
  validityState.value = coreElement.value.validity;
});

watch(
  validityState,
  (newValidity) => {
    if (!newValidity) return;
    emit("validityChange", newValidity);
  },
  { deep: true },
);
```

The `coreElement` is a ref so we should include it in the watchers on not rely on onMounted because the input element might change due to re-renders or future v-if conditions

Reply:
Also the `validityState` was not updated correctly before